os/net/lwip/src/api/sockets.c

```diff
@@ -1817,8 +1849,8 @@ static void event_callback(struct netconn *conn, enum netconn_evt evt, u16_t len
 			s = conn->socket;
 			SYS_ARCH_UNPROTECT(lev);
 		}
-
-		sock = get_socket(s);
+		pid = conn->pid;
+		sock = get_socket(s, pid);
```

Comment:
I think we don't need to use local variable, because `conn->pid` is only used once.

Reply:
modified